homeassistant/components/braviatv/entity.py

```diff
@@ -0,0 +1,39 @@
+"""A entity class for BraviaTV integration."""
+from typing import Any
+
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import BraviaTVCoordinator
+from .const import ATTR_MANUFACTURER, DOMAIN
+
+
+class BraviaTVEntity(CoordinatorEntity[BraviaTVCoordinator]):
+    """BraviaTV entity class."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: BraviaTVCoordinator,
+        unique_id: str,
+        model: str,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            manufacturer=ATTR_MANUFACTURER,
+            model=model,
+            name=f"{ATTR_MANUFACTURER} {model}",
+        )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+        await self.coordinator.async_turn_on()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+        await self.coordinator.async_turn_off()
```

Comment:
Please don't move this to base 😅 
I'm currently preparing a PR with a new backend library and later I want to add a `Button` platform for reboot and terminate apps. And these methods should not be in it.

Reply:
Sure, I'll revert this change.

BTW I can help with the library or testing if you want 😀

Reply:
> BTW I can help with the library or testing if you want

Sounds very nice! I'll let you know when it's ready :)